packages/medusa/src/api-v2/admin/promotions/[id]/route.ts

```diff
@@ -14,3 +15,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
 
   res.status(200).json({ promotion })
 }
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const updatePromotions = updatePromotionsWorkflow(req.scope)
+  const manager = req.scope.resolve("manager")
+  const promotionsData = [
+    {
+      id: req.params.id,
+      ...(req.validatedBody || {}),
+    },
+  ]
+
+  const { result, errors } = await updatePromotions.run({
+    input: { promotionsData },
+    context: { manager },
+    throwOnError: false,
+  })
+
+  if (Array.isArray(errors) && errors[0]) {
+    throw errors[0].error
+  }
```

Comment:
question: why does this handle errors this way when other's just rely on the throw from within the workflow?

Reply:
when the workflow-sdk handles the errors, it joins all existing error messages and throws a new raw error instance - [here](https://github.com/medusajs/medusa/blob/develop/packages/workflows-sdk/src/helper/workflow-export.ts#L128)

This way, we still keep the MedusaError context thrown by the modules and the API layer packages the error nicely.

Reply:
I see thanks

Reply:
And why is it that you are not using this approach in all endpoints then?

Reply:
missed out on the create, added them in there as well